File: Books.Core/BookDto.cs
namespace Grenis.AudioBooks.Core;

public class BookDto
{
    public int Id { get; init; }
    public string Title { get; init; } = string.Empty;
    public string? CustomTitle { get; init; }
    public string? Subtitle { get; init; }
    public string Author { get; init; } = string.Empty;
    public int? Year { get; init; }
    public string? ReadBy { get; init; }
    public int DurationSec { get; init; }
    public int ChapterCount { get; init; }
    public bool HasCover { get; init; }
    public string? Description { get; init; }
    public string? Publisher { get; init; }
    public string? Language { get; init; }
    public string? Isbn10 { get; init; }
    public string? Isbn13 { get; init; }
    public int? PageCount { get; init; }
    public double? Rating { get; init; }
    public int? RatingCount { get; init; }
    public string? GoogleBooksUrl { get; init; }
    public string? OpenLibraryUrl { get; init; }
    public DateTime AddedAt { get; init; }
    public int ProgressSec { get; init; }
    public bool IsCompleted { get; init; }
    public bool IsFavorite { get; set; }
    public DateTime? LastPlayedAt { get; init; }
    public int? ResumeChapterId { get; init; }
    public int ResumePositionSec { get; init; }
    public List<string> Genres { get; init; } = new();
    public bool IsStarted => LastPlayedAt is not null;
}

File: Mobile/Android/app/src/main/java/se/grenangen/audex/data/model/BookDto.kt
package se.grenangen.audex.data.model

import kotlinx.serialization.Serializable

@Serializable
data class BookDto(
    val id: Int,
    val title: String,
    val customTitle: String? = null,
    val subtitle: String? = null,
    val author: String,
    val year: Int? = null,
    val readBy: String? = null,
    val durationSec: Int,
    val chapterCount: Int,
    val hasCover: Boolean,
    val description: String? = null,
    val publisher: String? = null,
    val language: String? = null,
    val isbn10: String? = null,
    val isbn13: String? = null,
    val pageCount: Int? = null,
    val rating: Double? = null,
    val ratingCount: Int? = null,
    val openLibraryUrl: String? = null,
    val addedAt: String, // DateTime in C#
    val progressSec: Int,
    val isCompleted: Boolean,
    var isFavorite: Boolean,
    val lastPlayedAt: String? = null,
    val resumeChapterId: Int? = null,
    val resumePositionSec: Int,
    val genres: List<String>? = emptyList()
) {
    val isStarted: Boolean get() = lastPlayedAt != null
}
